### PR TITLE
Tweak: Update fontaweasome 6 [ED-4291]

### PR DESCRIPTION
**Pull request by guyc-E**

### CODE REVIEW

File: assets/lib/font-awesome/LICENSE.txt [hunk 1]
```diff
@@ -0,0 +1,165 @@
+Fonticons, Inc. (https://fontawesome.com)
```
Comment (matipojo) [inline assets/lib/font-awesome/LICENSE.txt:1]: Where is the preview license file?

File: assets/lib/font-awesome/css/fontawesome.min.css [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /*!
```
Comment (matipojo) [inline assets/lib/font-awesome/css/fontawesome.min.css:1]: What is the difference between the files with the `-` and without?
Why do we need both `fontawesome.css` and `font-awesome.css`

File: assets/lib/font-awesome/update-fontawesome.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?php
```
Comment (matipojo) [inline assets/lib/font-awesome/update-fontawesome.php:1]: Move it to a folder that will be removed on build.
Please consult with @KingYes
Currently, this file will be accessible from the web.

File: assets/lib/font-awesome/update-fontawesome.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+// How to update a version for Font Awesome:
+// FYI: "js" and "json" folders belongs to elementor and not fontawesome.
+
+// 1. Download the new version of Font Awesome.
+// 2. Replace the fontawesome folder excluding this file and migration folder.
+// 3. run this script.
+// 4. Files in js and json folders should be updated with the new icons.
+// 5. Delete redundant files in order to keep Elementor plugin size small.
+
+$metadata_path = 'metadata/icons.json';
```
Comment (matipojo) [inline assets/lib/font-awesome/update-fontawesome.php:11]: Please use absolute paths here and next
```suggestion
$metadata_path = __DIR__ . '/metadata/icons.json';
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/lib/font-awesome/LICENSE.txt (hunk 1 above)
Comment (guyc-E) [reply assets/lib/font-awesome/LICENSE.txt:1]: I don't understand, what do you mean by "preview license file"?

Comment (guyc-E) [reply assets/lib/font-awesome/LICENSE.txt:1]: Do you mean the license of font-awesome Pro? I will add it too.

File: assets/lib/font-awesome/css/fontawesome.min.css (hunk 2 above)
Comment (guyc-E) [reply assets/lib/font-awesome/css/fontawesome.min.css:1]: I can't find the file font-awesome.css, maybe I already deleted it.

File: core/upgrade/upgrades.php [hunk 5]
```diff
@@ -554,14 +554,14 @@ public static function _migrate_icon_fa4_value( $element, $args ) {
 				continue;
 			}
 
-			$element['settings'][ $new_name ] = Icons_Manager::fa4_to_fa5_value_migration( $element['settings'][ $old_name ] );
+			$element['settings'][ $new_name ] = Icons_Manager::fa4_to_fa6_value_migration( $element['settings'][ $old_name ] );
```
Comment (StyleShit) [inline core/upgrade/upgrades.php:557]: BC

File: includes/managers/icons.php [hunk 6]
```diff
@@ -378,7 +378,7 @@ public static function render_icon( $icon, $attributes = [], $tag = 'i' ) {
 	 *
 	 * @return array
 	 */
-	public static function fa4_to_fa5_value_migration( $value ) {
+	public static function fa4_to_fa6_value_migration( $value ) {
```
Comment (StyleShit) [inline includes/managers/icons.php:381]: BC

File: includes/managers/icons.php [hunk 7]
```diff
@@ -476,7 +476,7 @@ public function register_admin_settings( Settings $settings ) {
 						'' => esc_html__( 'No', 'elementor' ),
 						'yes' => esc_html__( 'Yes', 'elementor' ),
 					],
-					'desc' => esc_html__( 'Font Awesome 4 support script (shim.js) is a script that makes sure all previously selected Font Awesome 4 icons are displayed correctly while using Font Awesome 5 library.', 'elementor' ),
+					'desc' => esc_html__( 'Font Awesome 4 support script (shim.js) is a script that makes sure all previously selected Font Awesome 4 icons are displayed correctly while using Font Awesome 6 library.', 'elementor' ),
```
Comment (StyleShit) [inline includes/managers/icons.php:479]: %s

File: includes/managers/icons.php [hunk 8]
```diff
@@ -489,8 +489,8 @@ public function register_admin_tools_settings( Tools $settings ) {
 			'callback' => function() {
 				echo '<h2>' . esc_html__( 'Font Awesome Upgrade', 'elementor' ) . '</h2>';
 				echo '<p>' . // PHPCS - Plain Text
-				esc_html__( 'Access 1,500+ amazing Font Awesome 5 icons and enjoy faster performance and design flexibility.', 'elementor' ) . '<br>' . // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
-				esc_html__( 'By upgrading, whenever you edit a page containing a Font Awesome 4 icon, Elementor will convert it to the new Font Awesome 5 icon.', 'elementor' ) . // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+				esc_html__( 'Access 1,500+ amazing Font Awesome 6 icons and enjoy faster performance and design flexibility.', 'elementor' ) . '<br>' . // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+				esc_html__( 'By upgrading, whenever you edit a page containing a Font Awesome 4 icon, Elementor will convert it to the new Font Awesome 6 icon.', 'elementor' ) . // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```
Comment (StyleShit) [inline includes/managers/icons.php:493]: %s

File: includes/managers/icons.php [hunk 9]
```diff
@@ -509,7 +509,7 @@ public function register_admin_tools_settings( Tools $settings ) {
 							self::NEEDS_UPDATE_OPTION . '_upgrade',
 							wp_create_nonce( self::NEEDS_UPDATE_OPTION ),
 							esc_url( $this->get_upgrade_redirect_url() ),
-							esc_html__( 'Upgrade To Font Awesome 5', 'elementor' )
+							esc_html__( 'Upgrade To Font Awesome 6', 'elementor' )
```
Comment (StyleShit) [inline includes/managers/icons.php:512]: %s

File: includes/managers/icons.php [hunk 10]
```diff
@@ -543,14 +543,14 @@ public function get_upgrade_redirect_url() {
...
-	 * Ajax Upgrade to FontAwesome 5
+	 * Ajax Upgrade to FontAwesome 6
 	 */
 	public function ajax_upgrade_to_fa5() {
 		check_ajax_referer( self::NEEDS_UPDATE_OPTION, '_nonce' );
 
 		delete_option( 'elementor_' . self::NEEDS_UPDATE_OPTION );
 
-		wp_send_json_success( [ 'message' => esc_html__( 'Hurray! The upgrade process to Font Awesome 5 was completed successfully.', 'elementor' ) ] );
+		wp_send_json_success( [ 'message' => esc_html__( 'Hurray! The upgrade process to Font Awesome 6 was completed successfully.', 'elementor' ) ] );
```
Comment (StyleShit) [inline includes/managers/icons.php:553]: %s

File: includes/managers/icons.php [hunk 11]
```diff
@@ -543,14 +543,14 @@ public function get_upgrade_redirect_url() {
 	}
 
 	/**
-	 * Ajax Upgrade to FontAwesome 5
+	 * Ajax Upgrade to FontAwesome 6
 	 */
 	public function ajax_upgrade_to_fa5() {
```
Comment (StyleShit) [inline includes/managers/icons.php:548]: 6 + BC

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: assets/lib/font-awesome/css/font-awesome.css [hunk 12]
```diff
@@ -1,2337 +0,0 @@
-/*!
```
Comment (StyleShit) [inline assets/lib/font-awesome/css/font-awesome.css:1]: restore + minified

**Comment by hitsgee** [comment]
Any idea when this is going to be merged? Over a year now since this PR was created.

**Comment by leecollings-sb** [comment]
@yotamselementor Why was this closed?

**Comment by nicholaszein** [comment]
Hi @leecollings-sb,

This PR was closed because we are in the process of updating Font Awesome to version 6 in another PR that is more integrated with the current versions of Elementor and Elementor Pro, and more adapted to our infrastructure. 

We appreciate this contribution, and thank @guyc-E for taking the time to submit his PR.

Kind regards

**Pull request closed without merging**